src/components/select-option/select-option.tsx

```diff
@@ -0,0 +1,84 @@
+import { Component, h, Prop, EventEmitter, Event } from "@stencil/core";
+
+@Component({
+  tag: 'bds-select-option',
+  styleUrl: 'select-option.scss',
+  shadow: true
+})
+export class SelectOption {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  @Prop() value!: any;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  @Prop() label!: any;
+
+  /**
+   * The text value of the option.
+  */
+  @Prop() selected?= false;
+
+  /**
+   * If `true`, the user cannot interact with the select option.
+   */
+  @Prop() disabled?= false;
+
+  /** 
+   *  Quantity Description on option value, this item is locate to rigth in component.
+   */
+  @Prop() bulkOption?= '';
+
+  @Event() optionSelected: EventEmitter;
+
+
+  private onClickSelectOption = (): void => {
+    if (!this.disabled) {
+      this.optionSelected.emit({ label: this.label, value: this.value });
+    }
+  }
+
+  private attachOptionKeyboardListeners = (event: KeyboardEvent): void => {
+    const element = (event.target as HTMLElement);
+    const enterKey = 13;
```

Comment:
I think the keycodes may be all in uppercase and have a separate file to all keycodes.